### PR TITLE
fix(tests): Use  temporary directories for test git repositories

### PR DESCRIPTION
**Pull request by Natoandro**

### CODE REVIEW

**Review by Yohe-Am — APPROVED** [review]

File: typegate/tests/utils/shell.ts [hunk 1]
```diff
@@ -61,9 +65,16 @@ export async function shell(
 
   const { code, success } = await p.status;
 
+  const out = `-- start STDOUT --\n${res.stdout}\n-- end STDOUT --`;
```
Comment (zifeo) [inline typegate/tests/utils/shell.ts:68]: is this temp?

**Review by zifeo — COMMENTED** [review]
lgtm, I suppose the remaining failure is two typegraph re-using the same schema, we shall enforce this not to happen

File: typegate/tests/utils/test.ts [hunk 2]
```diff
@@ -52,8 +52,13 @@ function serve(typegate: Typegate, port: number): () => void {
 
 type MetaTestCleanupFn = () => void | Promise<void>;
 
+const defaultCli = await createMetaCli(shell);
+
 export class MetaTest {
   private cleanups: MetaTestCleanupFn[] = [];
+  shell = shell;
+  cli = defaultCli;
```
Comment (zifeo) [inline typegate/tests/utils/test.ts:60]: maybe t.meta is a name more coherent?

File: typegate/tests/utils/shell.ts (hunk 1 above)
Comment (Natoandro) [reply typegate/tests/utils/shell.ts:68]: Yes. I'm going to clean it up.

**Review by afmika — APPROVED** [review]

**Pull request merged**